sycl/source/detail/jit_compiler.cpp

```diff
@@ -278,7 +287,7 @@ static void resolveInternalization(ArgDesc &Arg, unsigned KernelIndex,
         // Recompute the local size for the previous definition with adapted
         // promotion target.
         auto NewPrevLocalSize =
-            getLocalSize(PreviousDefinition.NDRange,
+            getLocalSize(PreviousDefinition.NDRange, std::nullopt,
```

Comment:
Would `UserGlobalSize` ever be non-`std::nullopt` here?

Reply:
We could store it in the `PreviousDefinition` object, however in case the previous definition was in a RR kernel, then it won't have a local size, so the private-to-local-promotion upgrade won't be possible anyways.

Reply:
Okey, we should keep in mind we should probably change this in case we allow local promotion in `range` kernels